akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -209,6 +214,28 @@ private[akka] class SubFusingActorMaterializerImpl(
       context.asInstanceOf[ActorCell].removeFunctionRef(ref)
     case GetChildren =>
       sender() ! Children(context.children.toSet)
+    case GetChildrenSnapshots =>
+      // Arbitrary timeout but should always be quick, the failure scenario is that
+      // the child/stream stopped, and we do retry
+      implicit val timeout: Timeout = 1.second
+      implicit val scheduler = context.system.scheduler
+      val futureSnapshots: Iterable[Future[Option[StreamSnapshot]]] =
+        context.children.map(
+          child =>
+            retry(
+              { () =>
+                (child ? ActorGraphInterpreter.Snapshot).mapTo[StreamSnapshot].map(s => Some(s)).recover {
+                  case ex: AskTimeoutException
```

Comment:
can this fail for other reasons than timeout? if not, I don't see how the `retry` comes into the picture

Reply:
Yes, that'd be an actual timeout. Will rework and change.